pinot-common/src/main/java/org/apache/pinot/common/utils/FileUploadDownloadClient.java

```diff
@@ -1209,7 +1210,7 @@ public int downloadFile(URI uri, File dest, AuthProvider authProvider, List<Head
    */
   public File downloadUntarFileStreamed(URI uri, File dest, AuthProvider authProvider, List<Header> httpHeaders,
       long maxStreamRateInByte)
-      throws IOException, HttpErrorStatusException {
+      throws IOException, CompressorException, HttpErrorStatusException {
```

Comment:
Consider wrapping `CompressorException` into `IOException` or `RuntimeException` in the `TarCompressionUtils` so that we don't need to change caller code. I don't think it is useful to propagate this exception

Reply:
Wrapped `CompressorException` into `IOException`